packages/core-transaction-pool-mem/lib/connection.js

```diff
@@ -0,0 +1,340 @@
+'use strict'
+
+const Mem = require('./mem')
+const Storage = require('./storage')
+const container = require('@arkecosystem/core-container')
+const database = container.resolvePlugin('database')
+const emitter = container.resolvePlugin('event-emitter')
+const logger = container.resolvePlugin('logger')
+const { TRANSACTION_TYPES } = require('@arkecosystem/crypto').constants
+const { Transaction } = require('@arkecosystem/crypto').models
+const { TransactionPoolInterface } = require('@arkecosystem/core-transaction-pool')
+
+/**
+ * Transaction pool. It uses a hybrid storage - caching the data
+ * in memory and occasionally saving it to a persistent, on-disk storage (SQLite),
+ * every N modifications, and also during shutdown. The operations that only read
+ * data (everything other than add or remove transaction) are served from the
+ * in-memory storage.
+ */
+class TransactionPool extends TransactionPoolInterface {
+  /**
+   * Make the transaction pool instance. Load all transactions in the pool from
+   * the on-disk database, saved there from a previous run.
+   * @return {TransactionPool}
+   */
+  make () {
+    this.mem = new Mem()
+
+    this.storage = new Storage()
+
+    const allSerialized = this.storage.loadAllInInsertionOrder()
+    allSerialized.forEach(s => this.mem.add(new Transaction(s), this.options.maxTransactionAge, true))
+
+    return this
+  }
+
+  /**
+   * Disconnect from transaction pool.
+   * @return {void}
+   */
+  disconnect () {
+    this.__syncToPersistentStorage()
+    this.db.close()
+  }
+
+  /**
+   * Get the number of transactions in the pool.
+   * @return {Number}
+   */
+  getPoolSize () {
+    this.__purgeExpired()
+
+    return this.mem.getSize()
+  }
+
+  /**
+   * Get the number of transactions in the pool from a specific sender
+   * @param {String} senderPublicKey
+   * @returns {Number}
+   */
+  getSenderSize (senderPublicKey) {
+    this.__purgeExpired()
+
+    return this.mem.getIdsBySender(senderPublicKey).size
+  }
+
+  /**
+   * Add a transaction to the pool.
+   * @param {Transaction} transaction
+   */
+  addTransaction (transaction) {
+    if (this.transactionExists(transaction.id)) {
+      return logger.debug(
```

Comment:
While it works just fine and yields the expected `undefined`, let's not couple `return` with `logger`.